fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneSerializerTest.java

```diff
@@ -122,4 +121,52 @@ void testEncodingWithEncryption(boolean compressToo) throws Exception {
         final LuceneFileSystemProto.LuceneFileReference decryptedReference = LuceneFileSystemProto.LuceneFileReference.parseFrom(decodedValue);
         Assertions.assertEquals(content, decryptedReference.getContent());
     }
+
+    @ParameterizedTest
+    @MethodSource("encodedCompressedAndEncrypted")
+    void testProtobufMessageWithoutPrefix(boolean encode, boolean compress, boolean encrypt) throws Exception {
```

Comment:
Probably would be good to test the `FIeldInfos` as well - the structure is more flexible than `StoredFields` as all fields are optional

Reply:
That's a difference between `FieldInfo` and `StoredField`, not between `FieldInfos` and `LuceneStoredFields`. Both of the top-level messages have zero or more nested messages, which is what allows the determination.

Is it worth testing each one with the decoding as a black box? In that case, are there other interesting cases that you can think of?

Reply:
My concern was that we may interpret a non-empty protobuf as an encryption prefix. Having a non-empty `FieldInfos` with various types populated should be sufficient.
In reality we're likely OK since the first byte would always be `LEN` and the field number for the embedded message is always '1'.